src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -86,6 +86,23 @@ public int GetArrayLength()
             return _parent.GetArrayLength(_idx);
         }
 
+        /// <summary>
+        ///   Get the number of properties contained within the current object value.
+        /// </summary>
+        /// <returns>The number of properties contained within the current object value.</returns>
+        /// <exception cref="InvalidOperationException">
+        ///   This value's <see cref="ValueKind"/> is not <see cref="JsonValueKind.Object"/>.
```

Comment:
This would be the result of the `CheckExpectedType(JsonTokenType.StartObject, row.TokenType);` check, right? What does the `CheckValidInstance();` check validate?

Reply:
Similar to `ImmutableArray<T>` it's checking that it's not `default(JsonElement)`.

Reply:
The other JsonElement APIs don't appear to be documenting this failure mode either.